playground/backend/cmd/server/controller_test.go

```diff
@@ -494,19 +564,31 @@ func Test_processCode(t *testing.T) {
 				_, _ = lc.CreateExecutableFile(tt.code)
 			}
 
-			processCode(tt.args.ctx, cacheService, lc, exec, pipelineId, tt.args.env, tt.args.sdk)
+			if tt.cancelFunc {
+				go func(ctx context.Context, pipelineId uuid.UUID) {
```

Comment:
Why do we call it as goroutine?

Reply:
In this test, I want to set the cancel flag during the code processing (not before). So for that purpose, I need to wait sometime (2 seconds) while the `processCode` method starts to process code, and at this moment set the flag as true.